### PR TITLE
Milestone 2

### PR DESCRIPTION
**Pull request by SajjadAhmad14**
- Added components directory under src.
- Added containers directory under src.
- Added actions directory under src.
- Added reducers directory under src.
- Added reducers and actions.
- Created store.

### CODE REVIEW

File: src/components/App.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+import BookList from './BookList';
+import BooksForm from './BooksForm';
+import '../App.css';
+
+const App = () => (
+  <div className="App">
+    <Book />
+    <BooksForm />
+    <BookList />
+  </div>
```
Comment (rindrajosia) [inline src/components/App.js]: It only displays `BooksList` and `BookForm` components

File: src/components/Book.js [hunk 2]
```diff
@@ -0,0 +1,11 @@
+import React from 'react-dom';
+
+const Book = (book) => (
+  <div>
+    This is
+    {book}
+    component.
+  </div>
```
Comment (rindrajosia) [inline src/components/Book.js]: it represents single row of books with id, title and category [screenshot from Pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone2/Book/RepresentARowInBooksTable.png)

File: src/components/BookList.js [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+import BasicTable from './BookTable';
+
+const BookList = () => (
+  <div>
+    <BasicTable />
+  </div>
+);
+// const mapStateToProps = (state) => ({ id: state.id, title: state.title,
+// category: state.category });
+// export default connect(mapStateToProps, null)(BookList);
```
Comment (rindrajosia) [inline src/components/BookList.js]: Comments: It’s better to delete **commented piece of codes**. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

File: src/components/BookList.js [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import React from 'react';
+// import { connect } from 'react-redux';
```
Comment (rindrajosia) [inline src/components/BookList.js]: Comments: It’s better to delete **commented piece of codes**. If you need to review a previous version of your code, use git history ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/general/no_inline_comments.png) from [Pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/66243643/))

File: src/components/BookList.js [hunk 5]
```diff
@@ -0,0 +1,13 @@
...
+import BasicTable from './BookTable';
+
+const BookList = () => (
+  <div>
+    <BasicTable />
+  </div>
+);
```
Comment (rindrajosia) [inline src/components/BookList.js]: * it connects to Redux store, using the `connect()` function or [Redux hooks](https://react-redux.js.org/api/hooks), and fetches books from the state [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone2/BooksList/ConnetToReduxHooks.png) from [pathwright](https://microverse.pathwright.com/library/react-redux-v10/188148/path/step/107676536/)
* it uses `mapStateToProps` to fetch books from the `store`
* it displays books list in html table using `Book` component [screenshot from pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone2/BooksList/BooksAsTable.png)
* it passes a `book` object as a prop to each `Book` component

File: src/components/BookTable.js [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+              </TableCell>
+              <TableCell align="right">{row.title}</TableCell>
+              <TableCell align="right">{row.category}</TableCell>
+            </TableRow>
+          ))}
+        </TableBody>
+      </Table>
+    </TableContainer>
+  );
+}
```
Comment (rindrajosia) [inline src/components/BookTable.js]: I think, you don't need this file

File: src/index.js [hunk 7]
```diff
@@ -1,10 +1,23 @@
...
+import App from './components/App';
 
+const state = [
+  {
+    id: Math.random(),
+    title: 'book1',
+    category: 'Actions',
+  },
+];
+const store = createStore(state, booksReducer);
```
Comment (rindrajosia) [inline src/index.js]: It should be: 
`const store = createStore(booksReducer, state)`

File: src/reducers/books.js [hunk 8]
```diff
@@ -0,0 +1,5 @@
+const booksReducer = () => {
+
+};
+
+export default booksReducer;
```
Comment (rindrajosia) [inline src/reducers/books.js]: It is just an example, feel free to change it based on your src/actions/index.js:
```javascript
const books = (state, action) => {
  switch (action.type) {
    case CREATE_BOOK: {
      return {
        books: [...state.books, action.payload],
      };
    }
    case REMOVE_BOOK: {
      const { id } = action.payload;
      return {
        books: state.books.filter(book => book.id !== id),
      };
    }
    default:
      return state;
  }
};

```

**Review by rindrajosia — CHANGES_REQUESTED** [review]
Changes Required :red_circle:
To Highlight: :angel:

- The src/index.js imports `{ Provider }` from `react-redux` :+1: 
- App and Book components are placed in `src/components` directory :100: 

To improve :factory_worker: [REQUIRED]:

- [x] Comments: It’s better to delete commented piece of codes. If you need to review a previous version of your code, use git history (screenshot from Pathwright)
- **src/index.js:** 
- [x] it adds example books to the initial state object (please check my comment below :arrow_double_down: )
- [x] it wraps `App` in `Provider` and assign the whole thing into a variable:
```javascript
const wrappedApp = (
  <Provider store={store}>
    <App />
  </Provider>
);
```


- [x]  **`App` component:**  should only display `BooksList` and `BookForm` components
- 
- [x] **`Book` component:** should represent single row of books with id, title and category [screenshot from Pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone2/Book/RepresentARowInBooksTable.png)
- 
- **`BooksList` component:**
- [x] it is placed in `src/containers` directory
- [x] it connects to Redux store, using the `connect()` function or [Redux hooks](https://react-redux.js.org/api/hooks), and fetches books from the state [screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone2/BooksList/ConnetToReduxHooks.png) from [pathwright](https://microverse.pathwright.com/library/react-redux-v10/188148/path/step/107676536/)
- [x] it uses `mapStateToProps` to fetch books from the `store`
- [x] it displays books list in html table using `Book` component [screenshot from pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone2/BooksList/BooksAsTable.png)
- [x] it passes a `book` object as a prop to each `Book` component
- 
- [x] **`BookForm` component** : it is placed in `src/containers` directory
- 
- [x] **`index.js` reducer** : it combines app specific reducers (
[screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone2/ReducxReducers/CombinesAppReducers.png) from [pathwright](https://microverse.pathwright.com/library/fast-track-curriculum/69047/path/step/44896827/))
```javascript
import { combineReducers } from 'redux'
import books from './books'

export default combineReducers({
  books
})

```

- 
- [x] **`books.js` reducer** : it implements `CREATE_BOOK` and `REMOVE_BOOK` actions [screenshot from pathwright](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/react_reducx/bookstore/milestone2/ReducxReducers/CreateAndRemoveReducers.png)
- 
- [x] I got an error when I open your APP on my browser.


Nice to have [OPTIONAL] :

- [ ] 

That's it, you can submit another code review request after making these changes :recycle: :innocent:
With all my respect :pray:
Rindra Josia

File: src/components/App.js (hunk 1 above)
Comment (SajjadAhmad14) [reply src/components/App.js]: Done!

File: src/components/Book.js (hunk 2 above)
Comment (SajjadAhmad14) [reply src/components/Book.js]: Done!

File: src/components/BookList.js (hunk 5 above)
Comment (SajjadAhmad14) [reply src/components/BookList.js]: Done!

File: src/components/BookTable.js (hunk 6 above)
Comment (SajjadAhmad14) [reply src/components/BookTable.js]: Thanks for pointing it out!

**Review by Elbie-em — APPROVED** [review]
### STATUS: APPROVED ✅ 

Hi @SajjadAhmad14, awesome job implementing the changes requested by the previous TSE.

*All the best and keep up the good work* 💪 

> Elbie Moonga (Part-time TSE)

> Github (@Elbie-em)
> Slack (Elbie Moonga)
> Twitter ([@ElbieEm](https://twitter.com/ElbieEm))
> LinkedIn([Elbie Moonga](https://www.linkedin.com/in/elbiemoonga/))

**Pull request merged**